src/main/java/com/getbase/sync/Sync.java

```diff
@@ -52,12 +52,18 @@ public boolean fetch(BiPredicate<Meta, Map<String, Object>> predicate) {
         return fetchInternal(predicate);
     }
 
+    public SyncProcess newSyncProcess() {
+        return newSyncProcess(streamObservable);
+    }
+
     protected boolean fetchInternal(BiPredicate<Meta, Map<String, Object>> predicate) {
+        return newSyncProcess(predicate).run();
+    }
+
+    private SyncProcess newSyncProcess(BiPredicate<Meta, Map<String, Object>> predicate) {
         checkNotNull(predicate, "predicate parameter must not be null");
 
-        SyncProcess syncProcess = new SyncProcess(client, deviceUUID, sessionManager, predicate);
-        
-        return syncProcess.run();
+        return new SyncProcess(client, deviceUUID, sessionManager, predicate);
```

Comment:
Let's pass `timeout` as another param here. I believe we just need to alter existing ctor (no need to add another one).